lib/gringotts/gateways/authorize_net.ex

```diff
@@ -165,7 +157,8 @@ defmodule Gringotts.Gateways.AuthorizeNet do
   """
   @spec purchase(Float, CreditCard.t, Keyword.t) :: tuple
```

Comment:
Some of your specs are wrong/imprecise. This one can be:
```elixir
@spec purchase(float, CreditCard.t, Keyword.t) :: {:ok | :error, Response}
```
Pls refer the [typespec docs](https://hexdocs.pm/elixir/typespecs.html).